### DISCUSSION TITLE
Question about Chrome Web Store

### DISCUSSION THREAD

**Discussion by Pecacheu** [tsukasa/BdBrowser#23] [General]
Hi, just wanted to say thanks for making this and continuing to port BetterDiscord to the browser while the previous one was abandoned, it seems to be working pretty well for me on the latest Discord atm.

I was also just wondering how come this isn't actually published to the Chrome Web Store and has to be loaded in Developer Mode? Is there a specific reason like it's blocked by some weird policy, or is it just that you don't want to have to maintain a page for it? I have a few extensions published there so I do know how to do so if you don't know already.

**Comment by tsukasa** [comment]
Greetings,

thank you for your question and the kind words! As you already pointed out, the only reason you have to load the extension in developer mode is due to it not being on the Google Chrome Webstore. I hope the following three categories give you a bit of insight into why I do not have any plans to publish it on the storefront:

### Time and Support
- I maintain the extension primarily for myself. As a result, support is limited, updates are not guaranteed and having to deal with a new, potentially wider user base (read as: more novice-level users) is not something I desire due to simple time-management reasons.
- Using BdBrowser requires a certain level of knowledge from the user (i.e. when something goes awry because of a Discord update). Is it a plugin causing problems? Is it a BetterDiscord issue with the current web client? Or does the extension require an update due to internal changes in BetterDiscord? An average user will not know and, given the prior point about limited support, will probably have a bad user experience once things go south. Or even worse: Bother the BetterDiscord support staff about it.
- Just being on GitHub seems like a good filter to me. I do hope that people build their own version, compare the resulting files against the stock `dist` directory and see that there are no shenanigans or foul play going on.

### Morals and Ethics
- This is a fork of an already unofficial extension.
- Strencher, the original author, did not publish the extension onto the Webstore either. If anything were to end up on the Webstore, it should be published by the original author - or be something that the BetterDiscord project explicitly endorses.
- What eventually _did_ creep onto the Webstore in the past was a version of Strencher's codebase that opened ads and pulled a malicious remote payload. This led to the discontinuation of his work on the extension. When people refer to "the version on the Webstore", they mean that specific version.
- You will have 2-3 years of past negative association/baggage to deal with - a very undesirable starting position with a bad taste in many people's mouths.

### Technical Notes
- On a purely technical level, there should be no reason for Google to refuse the codebase (unless including the VfsTool.ps1 Powershell script is considered problematic). It should pass any check on Google's end and conforms to Google's latest requirements in terms of permissions usage.

**Reply by Pecacheu** [reply]
Thanks for the detailed reply! I can see your point, those reasons definitely do make sense even if the extension would technically be allowed on the web store. Also, I didn't know about the drama regarding a version getting published once that contained adware. I would certainly hate for that to happen to one of my projects, that's for sure!

Thank you for maintaining this port though, I love having themes on my Discord client but I've always preferred the web version vs. the bloated desktop app that runs a whole RAM-munching browser instance in the background of your machine 24/7 (even when the window is closed)

**Comment by joeyoropesa-dev** [comment]
One question

How we can remove those data too if we mess up with some BetterDiscord settings that broken some functions

![Screenshot_2023-06-13-18-43-03-186_com.github.android-edit.jpg](https://github.com/tsukasa/BdBrowser/assets/45666011/3afb56d6-1d11-44d3-bd2d-4b2804c57678)

**Reply by tsukasa** [reply]
If BetterDiscord does not load or even crash the page, you can format the virtual filesystem like this:
- In the extension options, set "Do not load BetterDiscord Renderer" to active.
- Navigate to Discord.com and open the web client, the regular vanilla web client should load.
- Open the Developer Tools by pressing F12 or going to Tools - Developer Tools.
- Be sure to be on the "Console" tab.
- To be on the safe side and/or to extract your configuration, create a VFS backup:
  ```javascript
  require("fs").exportVfsBackup();
  ```
- Issue the following command to wipe your virtual filesystem completely:
  ```javascript
  require("fs").formatVfs(true);
  ```
- In the extension options, set "Do not load BetterDiscord Renderer" to inactive.
- Reload your Discord.com web client tab.
- Your virtual filesystem should now be blank, and a new copy of the betterdiscord.asar will be downloaded during the page initialization.
